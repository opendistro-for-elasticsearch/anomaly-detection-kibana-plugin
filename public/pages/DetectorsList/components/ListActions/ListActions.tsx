/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { useState } from 'react';
import {
  EuiButton,
  EuiContextMenuItem,
  EuiContextMenuPanel,
  EuiFlexGroup,
  EuiFlexItem,
  EuiPopover,
} from '@elastic/eui';

interface ListActionsProps {
  onStartDetectors(): void;
  onStopDetectors(): void;
  onDeleteDetectors(): void;
  isActionsDisabled: boolean;
  isStartDisabled: boolean;
  isStopDisabled: boolean;
}

export const ListActions = (props: ListActionsProps) => {
  const [isOpen, setIsOpen] = useState(false);
  return (
    <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
      <EuiFlexItem grow={false} style={{ marginRight: '16px' }}>
        <EuiPopover
          id="actionsPopover"
          button={
            <EuiButton
              data-test-subj="listActionsButton"
              disabled={props.isActionsDisabled}
              iconType="arrowDown"
              iconSide="right"
              onClick={() => setIsOpen(!isOpen)}
            >
              Actions
            </EuiButton>
          }
          panelPaddingSize="none"
          anchorPosition="downLeft"
          isOpen={isOpen}
          closePopover={() => setIsOpen(false)}
        >
          <EuiContextMenuPanel>
            <EuiContextMenuItem
              key="startDetectors"
              data-test-subj="startDetectors"
              disabled={props.isStartDisabled}
              onClick={props.onStartDetectors}
            >
              Start
            </EuiContextMenuItem>

            <EuiContextMenuItem
              key="stopDetectors"
              data-test-subj="stopDetectors"
              disabled={props.isStopDisabled}
              onClick={props.onStopDetectors}
            >
              Stop
            </EuiContextMenuItem>

            <EuiContextMenuItem
              key="deleteDetectors"
              data-test-subj="deleteDetectors"
              onClick={props.onDeleteDetectors}
            >
              Delete
            </EuiContextMenuItem>
          </EuiContextMenuPanel>
        </EuiPopover>
      </EuiFlexItem>
    </EuiFlexGroup>
  );
};
