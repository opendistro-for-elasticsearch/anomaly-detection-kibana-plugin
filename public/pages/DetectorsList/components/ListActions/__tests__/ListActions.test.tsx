/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { fireEvent, render, wait } from '@testing-library/react';
import { ListActions } from '../ListActions';

describe('<ListActions /> spec', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  const defaultProps = {
    onStartDetectors: jest.fn(),
    onStopDetectors: jest.fn(),
    onDeleteDetectors: jest.fn(),
    detectors: [],
    isActionsDisabled: true,
    isStartDisabled: false,
    isStopDisabled: false,
  };
  describe('List actions', () => {
    console.error = jest.fn();
    test('renders component when disabled', () => {
      const { container } = render(<ListActions {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });
    test('renders component when enabled', () => {
      const { container } = render(
        <ListActions {...defaultProps} isActionsDisabled={false} />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test('should call onStartDetectors when clicking on start action', async () => {
      const { getByTestId } = render(
        <ListActions {...defaultProps} isActionsDisabled={false} />
      );
      fireEvent.click(getByTestId('listActionsButton'));
      await wait();
      fireEvent.click(getByTestId('startDetectors'));
      expect(defaultProps.onStartDetectors).toHaveBeenCalled();
    });
    test('should call onStopDetectors when clicking on stop action', async () => {
      const { getByTestId } = render(
        <ListActions {...defaultProps} isActionsDisabled={false} />
      );
      fireEvent.click(getByTestId('listActionsButton'));
      await wait();
      fireEvent.click(getByTestId('stopDetectors'));
      expect(defaultProps.onStopDetectors).toHaveBeenCalled();
    });
    test('should call onDeleteDetectors when clicking on delete action', async () => {
      const { getByTestId } = render(
        <ListActions {...defaultProps} isActionsDisabled={false} />
      );
      fireEvent.click(getByTestId('listActionsButton'));
      await wait();
      fireEvent.click(getByTestId('deleteDetectors'));
      expect(defaultProps.onDeleteDetectors).toHaveBeenCalled();
    });
  });
});
