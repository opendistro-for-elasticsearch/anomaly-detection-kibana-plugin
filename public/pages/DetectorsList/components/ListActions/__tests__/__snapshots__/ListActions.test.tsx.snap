// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListActions /> spec List actions renders component when disabled 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
    style="margin-right: 16px;"
  >
    <div
      class="euiPopover euiPopover--anchorDownLeft"
      id="actionsPopover"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary euiButton-isDisabled"
          data-test-subj="listActionsButton"
          disabled=""
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            <svg
              aria-hidden="true"
              class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonContent__icon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            />
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ListActions /> spec List actions renders component when enabled 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
    style="margin-right: 16px;"
  >
    <div
      class="euiPopover euiPopover--anchorDownLeft"
      id="actionsPopover"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary"
          data-test-subj="listActionsButton"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            <svg
              aria-hidden="true"
              class="euiIcon euiIcon--medium euiButtonContent__icon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.069 5.157L8.384 9.768a.546.546 0 01-.768 0L2.93 5.158a.552.552 0 00-.771 0 .53.53 0 000 .759l4.684 4.61c.641.631 1.672.63 2.312 0l4.684-4.61a.53.53 0 000-.76.552.552 0 00-.771 0z"
                fill-rule="non-zero"
              />
            </svg>
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
